From de7475dc9dd97e3897e4bd364df3f213440f9137 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 6 Nov 2021 09:46:24 +0100
Subject: [PATCH 094/107] Never use HTTP probes for connection detection

This patch disables the use of HTTP probes on Android < M devices
Connection will be always considered valid on such devices
---
 .../browser/net/connectivitydetector/ConnectivityDetector.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java b/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java
index c77eed5d19..f7c582c694 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java
@@ -180,7 +180,8 @@ public class ConnectivityDetector implements NetworkChangeNotifier.ConnectionTyp
 
         @Override
         public boolean shouldSkipHttpProbes() {
-            return false;
+            // always assume that connection is valid on Android < M
+            return true;
         }
     }
 
-- 
2.35.1

